packages/next/client/script.tsx

```diff
@@ -28,6 +28,39 @@ const ignoreProps = [
   'strategy',
 ]
 
+const injectPartytownSnippet = async () => {
+  const { enablePartytown, partytownConfig } = process.env
+    .__NEXT_OPTIMIZE_SCRIPTS as any
+
+  if (enablePartytown) {
+    if (
+      partytownConfig &&
+      !document.querySelector(`script[data-partytown-config]`)
+    ) {
+      // Only add partytown config if not already present from SSR
+      const scriptEl = document.createElement('script')
+      scriptEl.dataset.partytownConfig = ''
+      scriptEl.innerHTML = `partytown = ${JSON.stringify(partytownConfig)};`
+      document.head.appendChild(scriptEl)
+    }
+
+    if (!document.querySelector(`script[data-partytown]`)) {
```

Comment:
Should we just load partytown in SSR if enabled in config? Is the concern its not required in all pages? Whats the load size? Maybe we can bundle it and save the extra RTT and justify loading it everytime

Reply:
> Is the concern its not required in all pages?

Exactly. I was loading it in SSR *and* in CSR with the idea being that if there's no worker scripts on the initial page (`<script strategy="worker">`) but in a subsequent client-side transition; then load it only when necessary.

The snippet itself is just ~1KB though, so this was pretty unnecessary. I just removed this client-side logic completely. :)

> Maybe we can bundle it and save the extra RTT and justify loading it everytime

I did bundle it using `ncc` in the first iteration of the PR, but @adamdbradley (author of Partytown) provided some good feedback and mentioned that since it's still an experimental library—it might be better to allow users to install any version of the dependency themselves. In addition, the library's [publish size](https://packagephobia.com/result?p=%40builder.io%2Fpartytown) isn't trivial (328 KB) so it likely makes more sense for the user to install it only when needed.